hack/jenkins/e2e.sh

```diff
@@ -225,9 +225,16 @@ fi
 # TODO(zml): We have a bunch of legacy Jenkins configs that are
 # expecting junit*.xml to be in ${WORKSPACE} root and it's Friday
 # afternoon, so just put the junit report where it's expected.
-for junit in ${ARTIFACTS}/junit*.xml; do
-  ln -s ${junit} ${WORKSPACE}
-done
+results=($(ls "${ARTIFACTS}" | grep "junit.*.xml")) || true
+if [[ ! -z ${results:-} ]]; then
+    for junit in "${results[@]}"; do
+        # Jenkins runs that trigger sequentially (e.g. step1 -> step2 -> ...) will
+        # have e2e test results overwrite each other because they all run in the
+        # same workspace. For those cases, we prefix the final location with the
+        # step name.
+        mv "${ARTIFACTS}/${junit}" "${WORKSPACE}/${JENKINS_STEP_PREFIX:-}${junit}"
```

Comment:
1. What is the GCS uploader?

2. Wouldn't either setting a separate artifacts directory for each stage or allowing `e2e_test.go` to specify a different junit filename also both require env changes per Jenkins job? I'm wondering if there's any option that doesn't involve per-job Jenkins config (which we would lose if Jenkins died)? Seems like this PR and those two options all share that flaw.

3. With this PR we are no longer symlinking the junit files into the root workspace directory. We're just putting them there. Should we copy them there? Do we just not want them there?